[cms] download attachments

rhauckdot opened the pull request:
Added functionality to Options.jsx to accept dataAttachments prop from Viz config as either a string or array. Passed value is included into the JSZip call.

davelandry commented on packages/cms/src/components/Viz/Options.jsx:
```diff
@@ -115,9 +115,66 @@ class Options extends Component {
+          .then(response => {
+            if (response.status === 200 || response.status === 0) {
+              const blob = new Blob([response.data], {type: response.data.type});
+              return Promise.resolve(blob);
+            }
+            else {
+              return Promise.reject(new Error(response.statusText));
+            }
+          })
+          .then(data => {
+            zip.file(`${dAName}`, data, {binary: true});
+            zip.generateAsync({type: "blob"}).then(content => {
+              saveAs(content, `${filename(title)}.zip`);
+            });
+          });
+
+      }
+      else if (Array.isArray(dataAttachments)) {
+        // List of a files for attaching included in config
+        const requests = [];
+        for (let i = 0; i < dataAttachments.length; i++) {
+          requests[i] = axios.get(dataAttachments[i], {responseType: "blob"});
+        }
+
+        Promise.all(requests).then(responses => {
+
+          for (let i = 0; i < responses.length; i++) {
+            if (responses[i].status === 200 || responses[i].status === 0) {
+              // Pull data, grab file name, and add to ZIP
+              const blob = new Blob([responses[i].data], {type: responses[i].data.type});
+              const dAName = dataAttachments[i].split("/").pop();
+              zip.file(`${dAName}`, blob, {binary: true});
+            }
+          }
+          zip.generateAsync({type: "blob"})
+            .then(content => saveAs(content, `${filename(title)}.zip`));
+        });
+      }
+
+    }
```
@rhauckdot this can be written simpler, by coercing `dataAttachments` into an Array (even if it's only one item), and then running `Promise.all` on that array, something like this to prep:

```jsx
const attachmentURLs = dataAttachments ? Array.isArray(dataAttachments) ? dataAttachments : [dataAttachments] : [];
```

Then, you would just run your Array logic on `Promise.all(dataAttachments)`.

Another minor thing is to use a simple `forEach` instead of the `for` look inside of the then. The resulting code ends up a bit cleaner not having to juggle `i`:

```jsx
responses.forEach(response => {

});
```

rhauckdot commented:
Good call on converting to an array (I should have thought of that!). I added the requested documentation and converted the loops to forEach (old habit).
